jest.config.js

```diff
@@ -2,6 +2,7 @@ module.exports = {
     preset: 'ts-jest',
     testEnvironment: 'node',
     clearMocks: true,
+    bail: 1,
```

Comment:
not sure why, but jest registers as a success even on failure without this: https://github.com/PostHog/plugin-server/runs/3231307579

Unsure if there's a better way to fix the problem

Reply:
I'm guessing `bail` will produce the correct exit code but will not run any other files once a test in a file fails. While this is fine for CI, I wouldn't want this to be the case locally. We can maybe use `--bail` on the CI command rather than having this as a global config?

Also feels like this isn't the best way to fix this, but I'd be ok to merge as it's an improvement

Reply:
More context: It seems the failure doesn't register because it's a TS error rather than an actual error encountered when running the test.

Reply:
I believe the issue is `ts-jest`. We can try to compile before running tests and that might do it.